pkg/tagger/telemetry/telemetry.go

```diff
@@ -78,3 +81,54 @@ var (
 		[]string{}, "Number of of times the tagger has received a notification with a group of events",
 		telemetry.Options{NoDoubleUnderscoreSep: true})
 )
+
+// FetcherTelemetry stores telemetry counters for a single fetcher.
+type FetcherTelemetry struct {
+	Success  telemetry.SimpleCounter
+	NotFound telemetry.SimpleCounter
+	Error    telemetry.SimpleCounter
+}
+
+// NewFetcherTelemetry returns new instance of counters for the given fetcher name.
+func NewFetcherTelemetry(name string) FetcherTelemetry {
+	return FetcherTelemetry{
+		Success:  fetches.WithValues(name, FetchSuccess),
+		NotFound: fetches.WithValues(name, FetchNotFound),
+		Error:    fetches.WithValues(name, FetchError),
+	}
+}
+
+// CardinalityTelemetry contains queries counters for a single cardinality level.
+type CardinalityTelemetry struct {
+	EmptyEntityID telemetry.SimpleCounter
+	EmptyTags     telemetry.SimpleCounter
+	Success       telemetry.SimpleCounter
+}
+
+// NewCardinalityTelemetry creates new set of counters for a cardinality level.
+func NewCardinalityTelemetry(name string) CardinalityTelemetry {
+	return CardinalityTelemetry{
+		EmptyEntityID: queries.WithValues(name, QueryEmptyEntityID),
+		EmptyTags:     queries.WithValues(name, QueryEmptyTags),
+		Success:       queries.WithValues(name, QuerySuccess),
+	}
```

Comment:
`QueryEmptyEntityID`, `QueryEmptyTags` and `QuerySuccess` do not need to be exported.